config/jwt.js

```diff
@@ -0,0 +1,35 @@
+const jwt = require('jsonwebtoken'),
+  mongoose = require('mongoose'),
+  expiryDate = 86400, // 24hours
+  secret = process.env.SECRET,
+  User = mongoose.model('User');
+// routing process to authenticate users and generate token
+exports.authToken = (req, res) => {
+  // find the user
+  User.findOne({
+    email: req.body.email
+  }, (error, existingUser) => {
+    if (error) {
+      throw error;
+    }
+    if (!existingUser) {
+      return res.redirect('/#!/signup?error=notanexistinguser');
```

Comment:
this is a good implementation, however, the idea behind this task is to treat the server as an API, therefore, we do not want to have res.redirect, or res.set(a,b)

We want to return status codes and json responses only
e.g res.status(400).json({ error: 'Invalid Email/Password' }) // res.status(200).json({ user: {}, token: token })